File: LeetCode_Marathons/LCM4/Easy/Move_Zeroes/a.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int collision = 0;
        for (int i = 0; i <= n - 1; i++) {
            if (nums[i] != 0) {
                nums[i - collision] = nums[i];
                if (collision != 0) {
                    nums[i] = 0;
                }
            } else {
                collision++;
            }
        }
    }
};

File: LeetCode_Marathons/LCM4/Easy/Move_Zeroes/a.go
func moveZeroes(nums []int)  {
    n := len(nums)
    collision := 0
    for i := 0; i <= n - 1; i++ {
        if nums[i] != 0 {
            nums[i - collision] = nums[i]
            if collision != 0 {
                nums[i] = 0
            }
        } else {
            collision++
        }
    }
}